velox/type/Filter.cpp

```diff
@@ -145,15 +145,23 @@ bool BigintValuesUsingBitmask::testInt64Range(
   return !(min > max_ || max < min_);
 }
 
+template <typename T>
 BigintValuesUsingHashTable::BigintValuesUsingHashTable(
-    int64_t min,
-    int64_t max,
-    const std::vector<int64_t>& values,
+    T min,
+    T max,
+    const std::vector<T>& values,
     bool nullAllowed)
-    : Filter(true, nullAllowed, FilterKind::kBigintValuesUsingHashTable),
-      min_(min),
-      max_(max),
-      values_(values) {
+    : Filter(true, nullAllowed, FilterKind::kBigintValuesUsingHashTable) {
+  std::vector<T> hashTable;
+  if constexpr (std::is_same<T, int64_t>::value) {
```

Comment:
May be we can avoid `if check` by declaring min_ and max_ using type `T`.

Reply:
Thanks for the suggestion, templating this class would require changes in other classes as well because of functions like ```mergeWith``` and these changes seem tangential to the current PR. Would it be fine if the filter refactoring is taken up in a subsequent PR?